src/dynimage.rs

```diff
@@ -658,14 +658,22 @@ impl DynamicImage {
     }
 
     /// Resize this image using the specified filter algorithm.
-    /// Returns a new image. The image's aspect ratio is preserved.
     /// The image is scaled to the maximum possible size that fits
     /// within the bounds specified by `nwidth` and `nheight`.
+    /// The image's aspect ratio is preserved.
+    /// Returns a new image if the requested image size is different than the already existing one.
+    /// Otherwise the function returns a copy of self.
     pub fn resize(&self, nwidth: u32, nheight: u32, filter: imageops::FilterType) -> DynamicImage {
-        let (width2, height2) =
+        if (nwidth, nheight) == self.dimensions() {
+            // Image is already in the desired dimensions --> no need to resize.
+            return self.clone();
+        }
+        else {
```

Comment:
Could you remove this else block and just leave the rest of the method intact? Given that the if statement above contains a return, the behavior should be unchanged